### ISSUE TITLE
Int cannot represent non-integer value. Incorrect Int Scalar type.

### ISSUE THREAD

**Issue by TreeMan360** [neo4j/graphql#167]
**Describe the bug**
Trying to serialize count query I receive: `"Int cannot represent non-integer value: { low: 2000, high: 0 }"` where `2000` is the expected value. 

When running in Neo4J Browser and the equivalent query in GraphQL Architect the count is as expected. This currently appears to just fail when using `neo4j/graphql`.

**Type definitions**
```
import { gql } from 'apollo-server-express'

export const typeDefs = gql`
  type Event @exclude(operations: [CREATE, UPDATE, DELETE]) {
    id: ID! @id
    type: String!
    createdAt: DateTime! @timestamp(operations: [CREATE])
  }

  type TopEventResult @exclude {
    eventType: String!
    eventCount: Int!
  }

  type Query {
    topEvents(): [TopEventResult]!
      @cypher(
        statement: """
        MATCH (e:Event)
        WITH DISTINCT e.type AS eventType, COUNT(*) as eventCount
        RETURN { eventType: eventType, eventCount: eventCount }
        """
      )
  }
`
```

**To Reproduce**

1. Run a server with the following code
```
const app = express()

const startApolloServer = async () => {
  const driver = neo4j.driver(
    NEO4J_URL,
    neo4j.auth.basic(NEO4J_USERNAME, NEO4J_PASSWORD)
  )

  const neoSchema = new Neo4jGraphQL({
    typeDefs,
    resolvers,
    driver,
  })

  const server = new ApolloServer({
    schema: neoSchema.schema,
    context: ({ req }) => ({ req }),
  })

  await server.start()

  server.applyMiddleware({ app })

  await new Promise((resolve: (reason?: any) => void) =>
    app.listen({ port: 4000 }, resolve)
  )

  console.log(
    `🚀 Server ready at http://localhost:${PORT}${server.graphqlPath}`
  )

  return { server, app }
}

startApolloServer()
```

2. Then run the following Query...

```
{
  topEvents {
    eventType
    eventCount
  }
}
```
FWIW: Removing `eventCount` from the query works.

3. See error...
```
{
  "errors": [
    {
      "message": "Int cannot represent non-integer value: { low: 2000, high: 0 }",
      "locations": [
        {
          "line": 8,
          "column": 5
        }
      ],
      "path": [
        "topEvents",
        0,
        "eventCount"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "message": "Int cannot represent non-integer value: { low: 2000, high: 0 }",
          "stacktrace": [
            "GraphQLError: Int cannot represent non-integer value: { low: 2000, high: 0 }",
            "    at GraphQLScalarType.serializeInt [as serialize] (/var/task/node_modules/graphql/type/scalars.js:49:11)",
            "    at completeLeafValue (/var/task/node_modules/graphql/execution/execute.js:653:37)",
            "    at completeValue (/var/task/node_modules/graphql/execution/execute.js:578:12)",
            "    at completeValue (/var/task/node_modules/graphql/execution/execute.js:556:21)",
            "    at resolveField (/var/task/node_modules/graphql/execution/execute.js:472:19)",
            "    at executeFields (/var/task/node_modules/graphql/execution/execute.js:292:18)",
            "    at collectAndExecuteSubfields (/var/task/node_modules/graphql/execution/execute.js:748:10)",
            "    at completeObjectValue (/var/task/node_modules/graphql/execution/execute.js:738:10)",
            "    at completeValue (/var/task/node_modules/graphql/execution/execute.js:590:12)",
            "    at /var/task/node_modules/graphql/execution/execute.js:620:25"
          ]
        }
      }
    }, ...
}
```

**Expected behavior**
Should have returned the count value of 2000 rather than trying to parse `{ low: 2000, high: 0 }`.

**System (please complete the following information):**
-   OS: macOS Big Sur 11.1
-   Version: @neo4j/graphql@1.0.0-beta.2
-   Node.js version: 12.20.1

**Additional context**
@darrellwarde stated on a discussion on Discord in the #graphql channel:
> "The only thing that really stands out to me here is that the error is being thrown from the default GraphQL Int scalar, but we have our custom scalar which would be located at /var/task/node_modules/@neo4j/graphql/dist/schema/scalars/Int.js on your machine. So the error itself is valid, the unusual behaviour is that our custom scalar isn't being used... Those are my initial thoughts"

**Comment by danstarns** [comment]
Hey @TreeMan360. Thanks for reporting! We have spent some time looking into this and can't replicate it. In the latest release do you still experience this?

**Comment by darrellwarde** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Scalar problem](https://trello.com/c/4qm1yKU5/21-scalar-problem)

**Comment by Pitchlab** [comment]
I managed to work around this using 

```
const driver = neo4j.driver(
    "bolt://localhost:7687",
    neo4j.auth.basic(username, password),
    { disableLosslessIntegers: true }           
);
```

Based on the [following post](https://github.com/neo4j/neo4j-javascript-driver#numbers-and-the-integer-type)

**Comment by darrellwarde** [comment]
> I managed to work around this using
> 
> ```
> const driver = neo4j.driver(
>     "bolt://localhost:7687",
>     neo4j.auth.basic(username, password),
>     { disableLosslessIntegers: true }           
> );
> ```
> 
> Based on the [following post](https://github.com/neo4j/neo4j-javascript-driver#numbers-and-the-integer-type)

This can be a potential fix for this, but do be aware that `BigInt` values are unlikely to work with this option switched on. We are going to be looking at a more holistic fix for this very soon.

**Comment by oskarhane** [comment]
@TreeMan360 Hi, we're still having issues reproducing this. I've copied your setup above and it works well for me.
Could you create a separate repo where this behavior is seen, to help us dig into it?

**Comment by oskarhane** [comment]
@Pitchlab Do you have an example where you experience this issue, that you can share with us?
It'd be super helpful.

**Comment by Pitchlab** [comment]
If i don't use the config, any int field that I try to use with graphql
shows up as something like {low: 100, high:0} and gives error messages that
these are not valid javascript ints.

**Comment by tlester** [comment]
disableLosslessIntegers didn't work for me.  Still got the "Invalid input. '1.0' is not a valid value. Must be a non-negative integer." error.   

```
const driver = neo4j.driver(NEO4J_URI, neo4j.auth.basic(NEO4J_USER, NEO4J_PASS), {
    disableLosslessIntegers: true,
})
```

**Comment by darrellwarde** [comment]
A fix (hopefully!) for this has now been merged, and will be released shortly (aiming for tomorrow). Please do test and feel free to reopen this ticket if it hasn't addressed your problems! 🙂

**Comment by darrellwarde** [comment]
After much back and forth of thinking we had fixed this bug, it should finally be _fully_ fixed by #297. 🤞  Apologies it took a while to get to this stage, it was a tricky bug to find but then very clear once we had found it!
